src/query/service/src/pipelines/processors/transforms/aggregator/transform_group_by_partial.rs

```diff
@@ -149,6 +149,10 @@ impl<Method: HashMethodBounds> AccumulatingTransform for TransformPartialGroupBy
     const NAME: &'static str = "TransformPartialGroupBy";
 
     fn transform(&mut self, block: DataBlock) -> Result<Vec<DataBlock>> {
+        if block.is_empty() {
```

Comment:
I think you did not find the key reason for this bug.

Why the tests failed when you enable enable_experimental_aggregate_hashtable yet it succeeded without this config ?   There must be some bug inside the new agg.